docs_src/design/adr/0026-Common Configuration.md

```diff
@@ -0,0 +1,401 @@
+# Common Configuration
+### Submitters
+- Lenny Goodell (Intel)
+- Elizabeth Lee (Intel)
+
+## Change Log
+- [pending](URL of PR TBD) (2022-11-29)
+
+## Referenced Use Case(s)
+- [Common Configuration UCR](URL of use case link TBD)
+
+## Context
+This ADR describes the architecture of the new common configuration capability which impacts all services. Requirements for this new capability are described in the above referenced UCR. This is deemed architecturally significant due to the cross-cutting impacts.
+
+### Current Design
+
+The following flow chart demonstrates the bootstrapping of each services' configuration in the current Levski release.
+
+![](common-config-images/EdgeX 2.x Configuration bootstrapping flowchart.png)
+
+## Proposed Design
+The configuration settings that are common to all services will be partitioned out into a separate common configuration source. This  common configuration source will be pushed into the Configuration Provider by a new `core-common-config` service. When the Configuration Provider is not used the `core-common-config` service will provide the common configuration via a RESTful endpoint.
+
+During bootstrapping, each service will either load the common configuration from the Configuration Provider or via URI to some endpoint that provides the common configuration. This could be the new service's REST endpoint. Each service will have additional private configuration, which may overwrite and/or extend the common configuration.  
+
+An additional common configuration setting must be present to indicate all other common settings have been pushed to the Configuration Provider. This setting is stored last and the services must wait for this setting to be present prior to pulling the common settings.
+
+### Configuration Bootstrapping
+
+The following flow chart demonstrates the bootstrapping of each services' configuration with this new common configuration capability.
+
+![](common-config-images/EdgeX 3.x Configuration bootstrapping flowchart.png)
+
+### Secret Store Configuration
+
+As part of this design, the Secret Store configuration is being removed from the service configuration (common and private). This is so the Secret Provider can be instantiated prior to processing the service's configuration which may require the Secret Provider. The Secret Store configuration will now be a combination of default values and environment variable overrides. These environment variables will be the same as the ones that are currently used to override the configuration.
+
+### Specifying the Common Configuration location
+
+If the `-cp/--configProvider` command line option is used, the service will default to pulling the common configuration from a standard path in the Configuration Provider. i.e. `edgex/3.0/common/` The `-cp/--configProvider` option assumes the usage of the core-common-config service and cannot be used with the `-cc/--commonConfig` option.

```

Comment:
```suggestion
If the `-cp/--configProvider` command line option is used, the service will default to pulling the common configuration from a standard path in the Configuration Provider i.e. `edgex/3.0/common/`. In this case, the value of `-cc/--commonConfig` option is ignored.
```

Reply:
I think it would be better to **error out** if both `-cp` and `-cc` are specified. Otherwise, the user will not know which option was used.

Reply:
Yes, we could error out.